### PR TITLE
Optimise video upload handling and processing to be 100% async

### PR DESCRIPTION
**Pull request by prologic**
##### Summary

ssia

##### Component Name

- area/backend
- area/ui

##### Test Plan

Backend parts are tested locally and work as expected with appropriate error handling.

TBD:

- [x] JS code for handling on the UI side (possibly the Flutter app can reuse?)
- [x] Full e2e UI testing

##### Additional Information

This avoids completely the CF (_default_) 100s timeout and allows uploads of
media requiring transcoding to be uploaded up to the maximum upload limits
of the pod.

### CODE REVIEW

File: internal/handlers.go [hunk 1]
```diff
@@ -2209,55 +2209,96 @@ func (s *Server) UploadMediaHandler() httprouter.Handle {
...
+
+		t, ok := s.tasks.Lookup(uuid)
+		if !ok {
+			log.Warnf("no task found by uuid: %s", uuid)
+			http.Error(w, "Task Not Found", http.StatusNotFound)
+			return
+		}
+
+		data, err := json.Marshal(t.Result())
+		if err != nil {
```
Comment (dooven) [inline internal/handlers.go:2297]: Should we log the error here

File: internal/utils.go [hunk 2]
```diff
@@ -838,31 +832,39 @@ func StoreUploadedVideo(conf *Config, f io.Reader, resource, name string, opts *
...
 
-	wg.Add(3)
+	wg.Add(4)
+
 	go TranscodeWebM(ctx, errs)
 	go TranscodeMP4(ctx, errs)
 	go GeneratePoster(ctx, errs)
 
+	go func(ctx context.Context) {
+		defer wg.Done()
```
Comment (dooven) [inline internal/utils.go:845]: Will this wg.Done() gets called 4 times?

File: internal/handlers.go (hunk 1 above)
Comment (prologic) [reply internal/handlers.go:2297]: Normally I would but these types of errors (_marshalling errors_) are rare at best and should not happen at all.

File: internal/utils.go (hunk 2 above)
Comment (prologic) [reply internal/utils.go:845]: It does! This is how `sync.WaitGroup` works. You call `wg.Add(4)` then `wg.Done()` 4 times. Internally it has an atomic counter.

Comment (dooven) [reply internal/utils.go:845]: I see. I think it should be called in the select right? That way it gets called multiple times.
```go
	go func(ctx context.Context) {
		select {
		case err := <-errs:
			log.WithError(err).Errorf("TranscodeVideo() error")
			errors++
			wg.Done()
		case <-ctx.Done():
			wg.Done()
			return
		}
	}(ctx)
```

Comment (prologic) [reply internal/utils.go:845]: The function/goroutine that collects the errors actually has a `defer wg.Done()` so it does the right thing already. You don't want to call `wg.Done()` more times than you've called `wg.Add(delta)` But I'l double check the code 👍

Comment (dooven) [reply internal/utils.go:845]: Oh I see where it's getting called now. Sorry I missed it. Thanks for the clarification!

**Review by dooven — APPROVED** [review]

**Pull request merged**